### PR TITLE
fix(manager/gitlabci): name is not directly followed by image tag

### PR DESCRIPTION
**Pull request by secustor**
## Changes:
Loops over all lines until a name is matched or a tag on the same level is found 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/gitlabci/extract.ts [hunk 1]
```diff
@@ -7,12 +7,17 @@ import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
 import type { GitlabPipeline } from './types';
 import { replaceReferenceTags } from './utils';
 
+const commentsRe = /^\s*#/;
+const whitespaceRe = /^(?<whitespace>\s*)/;
+const imageRe = /^(?<whitespace>\s*)image:\s*'?"?(?<image>[^\s'"]+|)'?"?\s*$/;
+const nameRe = /^\s*name:\s*'?"?(?<depName>[^\s'"]+|)'?"?\s*$/;
+const serviceRe = /^\s*-\s*(?:name:\s*)?'?"?(?<depName>[^\s'"]+)'?"?\s*$/;
```
Comment (viceice) [inline lib/manager/gitlabci/extract.ts:14]:
```suggestion
const imageRe = /^(?<whitespace>\s*)image:\s+'?"?(?<image>[^\s'"]+|)'?"?\s*$/;
const nameRe = /^\s*name:\s+'?"?(?<depName>[^\s'"]+|)'?"?\s*$/;
const serviceRe = /^\s*-\s*(?:name:\s+)?'?"?(?<depName>[^\s'"]+)'?"?\s*$/;
```

I think at least one whitespace is required after a yaml key

File: lib/manager/gitlabci/extract.ts [hunk 2]
```diff
@@ -7,12 +7,17 @@ import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
 import type { GitlabPipeline } from './types';
 import { replaceReferenceTags } from './utils';
 
+const commentsRe = /^\s*#/;
+const whitespaceRe = /^(?<whitespace>\s*)/;
+const imageRe = /^(?<whitespace>\s*)image:\s*'?"?(?<image>[^\s'"]+|)'?"?\s*$/;
```
Comment (viceice) [inline lib/manager/gitlabci/extract.ts:12]:
```suggestion
const imageRe = /^(?<whitespace>\s*)image:\s+'?"?(?<image>[^\s'"]+|)'?"?\s*$/;
```

You've lost one. 😏 Next time you can simply apply my sugesstion 😉

Comment (secustor) [reply lib/manager/gitlabci/extract.ts:12]: I have applied your suggestion :P 
but this regex has to match this case too: 
```yaml
image:
  name: hub.docker.io/myOrg/myImage
```

Comment (viceice) [reply lib/manager/gitlabci/extract.ts:12]: It won't match that anyways, as we go line-by-line.

Comment (viceice) [reply lib/manager/gitlabci/extract.ts:12]: ok, what about the updated regex above 🤔

Comment (secustor) [reply lib/manager/gitlabci/extract.ts]: That didn't worked either, but the problem has been in another section.    
I wanted initially to change the regexes as little as possible, but as we have started to optimize them I have changed the others as well. 

Regarding the image regex I have gone with this: 
```
const imageRe = /^(?<whitespace>\s*)image:(?:\s+['"]?(?<image>[^\s'"]+)['"]?)?\s*$/;
```

File: lib/manager/gitlabci/extract.ts [hunk 3]
```diff
@@ -7,12 +7,18 @@ import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
 import type { GitlabPipeline } from './types';
 import { replaceReferenceTags } from './utils';
 
+const commentsRe = /^\s*#/;
+const whitespaceRe = /^(?<whitespace>\s*)/;
+const imageRe =
+  /^(?<whitespace>\s*)image:(?:\s+['"]?(?<image>[^\s'"]+)['"]?)?\s*$/;
+const nameRe = /^\s*name:\s+['"]?(?<depName>[^\s'"]+|)['"]?\s*$/;
```
Comment (viceice) [inline lib/manager/gitlabci/extract.ts:14]:
```suggestion
const nameRe = /^\s*name:\s+['"]?(?<depName>[^\s'"]+)['"]?\s*$/;
```

What does the `|` match? 🤔

Comment (secustor) [reply lib/manager/gitlabci/extract.ts:14]: Nothing, I guess. I have removed them from the image regex, but it seems I have missed that one

**Review by viceice — APPROVED** [review]
looks much cleaner now 🎉

**Pull request merged**